### DISCUSSION TITLE
deepObserve vs toJS

### DISCUSSION THREAD

**Discussion by TechAkayy** [mobxjs/mobx#3208] [Q&A] [answered]
Hi all, I'm new to Mobx. 

I'm trying to get my head around the difference between mobx-utils/deepObserve and using toJS in autorun. toJS should let us detect all the changes to an object. How is deepObserve different and in what instance? 

```
import { deepObserve } from "mobx-utils";
import { observable, action, toJS, autorun } from "mobx";

const myList = observable({
    fruits: {
        name: "apple",
        quantity: 10,
        varities: [
            {
                color: "green",
            },
            {
                color: "red",
            },
        ],
    },
});

deepObserve(myList, (change, path, root) => {
    console.log(myList);
    // console.log(change);
});

autorun(() => {
    console.log(toJS(myList));
});

const myAction = action((myList) => {
    myList.fruits.varities[0].color = "orange";
});

myAction(myList);
```

**Comment by urugator** [comment, marked as answer]
If you modify multiple properties in `action` it will result in multiple invocations of `deepObserve` vs single invocation of `autorun`.

Could help: https://github.com/mobxjs/mobx/issues/1207#issuecomment-337647784

**Reply by TechAkayy** [reply]
Thanks a bunch!

**Comment by mweststrate** [comment]
also toJS deepclones clones the entire object, which can be overkill and
even confusing, unless you need a deep defensive copy.
